test/guardpage.c

```
@@ -0,0 +1,397 @@
+  unsigned char *pg;
+
+  if (chunk == NULL) {
+    return NULL;
+  }
+  memset(chunk, 0xAB, want);
+
+  /* First page boundary at least one page into our own chunk. */
+  pg = (unsigned char *)((((uintptr_t)chunk + page_size - 1)
+                          & ~(uintptr_t)(page_size - 1)) + page_size);
+  if (pg + page_size > chunk + want) {
+    return NULL;
```

Cppcheck (2.21.0)

[error] Memory leak

(memleak)